### PR TITLE
Add cookiecutter-pylibrary skeleton

### PR DESCRIPTION
**Pull request by nealmcb**
Provide a skeleton for library development, generated by
[ionelmc/cookiecutter\-pylibrary: Enhanced cookiecutter template for Python libraries\.](https://github.com/ionelmc/cookiecutter-pylibrary)

Todo:
Enable and test various integrations:
* [x] readthedocs [set up]
* [x] travis-ci [set up, need to merge PR to test it]
* [x] codecov [uses travis]
* [ ] requires.io setup and fix image 

Review changes
* [x] Test `tox` command, pick right set of build targets.
* [x] Review .gitignore changes and merge as necessary
* [ ] Drop README.rst skeleton and replace with our own? In Markdown?
* [ ] Do we want to maintain a CHANGELOG.rst?
* [ ] Do we want AUTHORS.rst? What to list there?
* [ ] Use setuptools_scm instead of bumpversion?

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    author='R2B2 Team',
+    author_email='r2b2@example.org',
+    url='https://github.com/gwexploratoryaudits/r2b2',
+    packages=find_packages('src'),
+    package_dir={'': 'src'},
+    py_modules=[splitext(basename(path))[0] for path in glob('src/*.py')],
+    include_package_data=True,
+    zip_safe=False,
+    classifiers=[
+        # complete classifier list: http://pypi.python.org/pypi?%3Aaction=list_classifiers
```
Comment (smorin8674) [inline setup.py]: Do we need to update this for Pytohn 3.7 only? Also is Production/Stable the correct status?

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -0,0 +1,90 @@
+============
```
[the remark this answers is not in the document]
Comment (nealmcb) [reply CONTRIBUTING.rst]: Agreed. But I'd prefer to do that in future commits.

File: CHANGELOG.rst [hunk 3]
```diff
@@ -0,0 +1,8 @@
+
+Changelog
+=========
+
+0.1.0 (2020-02-11)
+------------------
+
+* First release on PyPI.
```
[the remark this answers is not in the document]
Comment (nealmcb) [reply CHANGELOG.rst]: Done

File: setup.py (hunk 1 above)
Comment (nealmcb) [reply setup.py]: Good catch.

File: ci/templates/.travis.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
+language: python
+dist: xenial
+cache: false
+env:
+  global:
+    - LD_PRELOAD=/lib/x86_64-linux-gnu/libSegFault.so
+    - SEGFAULT_SIGNALS=all
+matrix:
+  include:
+    - python: '3.6'
```
[the remark this answers is not in the document]
Comment (nealmcb) [reply ci/templates/.travis.yml]: I think so. I'll push an update and see if it works.

**Pull request merged**